app/javascript/components/common/CopyToClipboardButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react'
+
+export function CopyToClipboardButton({ textToCopy }: { textToCopy: string }) {
+  const [justCopied, setJustCopied] = useState(false)
+
+  const onClick = async () => {
```

Comment:
Maybe we can use `useCallback` here?

Reply:
I'm not sure. I've tried using it but failed to get it working. It looked like the `useCallback` memoization prevented the code from being updated in subsequent calls, which meant that the clipboard wasn't updated after the first time.

Reply:
Did you add `textToCopy` to the dependencies of `useCallback` ?